File: OrdersService/Orders.Domain/Entities/Order.cs
using Orders.Domain.Enums;

namespace Orders.Domain.Entities;

public class Order
{
    public int Id { get; set; }
    public int ProductId { get; set; }

    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }

    public OrderStatus Status { get; set; }
    public Product Product { get; set; }
}

File: shop-domain/src/models/Order.ts
import ModelBase from "./ModelBase";
import Product from "./Product";

export default class Order extends ModelBase {
    productId!: string;
    quantity: number = 1;
    product: Product | null = null;
    unitPrice: number = 0;
    /**
     *
     */
    constructor(obj: unknown | Order | null = null) {
        super();
        if (obj) {
            Object.assign(this, obj);
        }
    }
}